integration_tests/src/payments_althea.rs

```diff
@@ -1,2 +1,125 @@
-pub fn run_althea_payments_test_scenario() {}
+use std::thread;
+use std::time::Duration;
+
+use crate::five_nodes::five_node_config;
+use crate::setup_utils::database::start_postgres;
+use crate::setup_utils::namespaces::*;
+use crate::setup_utils::rita::thread_spawner;
+use crate::utils::{
+    althea_system_chain_client, althea_system_chain_exit, generate_traffic, get_default_settings,
+    print_althea_balances, query_debts, register_to_exit, send_althea_tokens, test_reach_all,
+    test_routes, NODE_IP,
+};
+use althea_types::ALTHEA_PREFIX;
+use deep_space::Address as AltheaAddress;
+use deep_space::{EthermintPrivateKey, PrivateKey};
+use log::info;
+
+const USDC_TO_WEI_DECIMAL: u64 = 1_000_000_000_000u64;
+
+const ALTHEA_EVM_PRIV_BYTES: &str =
+    "3b23c86080c9abc8870936b2eb17ecb808f5ad3b318018b3e23873013379e4d6";
+
+pub fn get_althea_evm_priv() -> EthermintPrivateKey {
+    ALTHEA_EVM_PRIV_BYTES
+        .parse::<clarity::PrivateKey>()
+        .unwrap()
+        .into()
+}
+
+pub fn get_althea_evm_pub() -> AltheaAddress {
+    get_althea_evm_priv().to_address(ALTHEA_PREFIX).unwrap()
+}
+
+pub fn get_althea_grpc() -> String {
+    format!("http://{}:9091", NODE_IP)
+}
+
+pub async fn run_althea_payments_test_scenario() {
+    info!("Starting althea payments test");
+
+    let node_config = five_node_config();
+    let namespaces = node_config.0;
+    let expected_routes = node_config.1;
+
+    let (rita_settings, rita_exit_settings) = get_default_settings();
+
+    namespaces.validate();
+    start_postgres();
+
+    let res = setup_ns(namespaces.clone());
+    info!("Namespaces setup: {res:?}");
+
+    // Modify configs to use Althea chain
+    let rita_exit_settings = althea_system_chain_exit(rita_exit_settings);
+    let rita_settings = althea_system_chain_client(rita_settings);
+
+    let rita_identities = thread_spawner(namespaces.clone(), rita_settings, rita_exit_settings)
+        .expect("Could not spawn Rita threads");
+    info!("Thread Spawner: {res:?}");
+
+    // Test for network convergence
+    test_reach_all(namespaces.clone());
+
+    test_routes(namespaces.clone(), expected_routes);
+
+    info!("Registering routers to the exit");
+    for r in namespaces.names.clone() {
+        if let NodeType::Client = r.node_type {
+            let res = register_to_exit(r.get_name()).await;
+            if !res.is_success() {
+                panic!("Failed to register {} to exit with {:?}", r.get_name(), res);
+            } else {
+                info!("{} registered to exit", r.get_name());
+            }
+        }
+    }
+
+    thread::sleep(Duration::from_secs(10));
+
+    let mut from_node: Option<Namespace> = None;
+    let mut forward_node: Option<Namespace> = None;
+    let mut end_node: Option<Namespace> = None;
+    // top up routers
+    for ns in namespaces.names.clone() {
+        info!("Namespace : {:?}", ns);
+        if ns.id == 1 {
+            from_node = Some(ns.clone());
+        }
```

Comment:
I think there's a utility function for grabbing namespaces by id that could replace this.

Reply:
done